python/ray/serve/tests/test_schema.py

```diff
@@ -352,23 +351,6 @@ def test_route_prefix(self):
         deployment_schema["route_prefix"] = None
         DeploymentSchema.parse_obj(deployment_schema)
 
-    def test_mutually_exclusive_num_replicas_and_autoscaling_config(self):
-        # num_replicas and autoscaling_config cannot be set at the same time
```

Comment:
Do we need to replace this test with a similar test?  Or are we no longer enforcing this condition?

Reply:
What's the behavior if the user provides `num_replicas` and `autoscaling_config` at the same time? Does the deployment start at `num_replicas` and then scale up or down?

Reply:
default behavior is num_replicas will be overwritten by min_replicas.  https://github.com/ray-project/ray/blob/master/python/ray/serve/controller.py#L384

num_replicas is already overused unfortunately, the function you mention should be init_num_replicas dedicate for autoscaling. (we don't have it yet)

Reply:
I see– could we keep the restriction in that case? I think it makes sense to loudly raise an error if the user provides `num_replicas` and `autoscaling_config` at the same time since `num_replicas` will be ignored.

Reply:
it is not provided by user. When we do bind(), we will copy the deployment https://github.com/ray-project/ray/blob/master/python/ray/serve/deployment.py#L192
the num_replicas will be set to 1, because that is the default value. So it will always cause num_replicas and autoscale_config be set together

Reply:
Ah, got it, thanks for explaining!

Reply:
> we will copy the deployment https://github.com/ray-project/ray/blob/master/python/ray/serve/deployment.py#L192
the num_replicas will be set to 1,

We should change this behavior instead of allowing conflicting options.

Reply:
added it back